cores/esp8266/HardwareSerial.h

```diff
@@ -135,16 +135,45 @@ class HardwareSerial: public Stream
         // return -1 when data is unvailable (arduino api)
         return uart_peek_char(_uart);
     }
+
+    virtual bool peekBufferAPI () const override
+    {
+        return true;
+    }
+
+    // return a pointer to available data buffer (size = available())
+    // semantic forbids any kind of read() before calling peekConsume()
+    const char* peekBuffer () override
+    {
+        return uart_peek_buffer(_uart);
+    }
+
+    // return number of byte accessible by peekBuffer()
+    size_t availableForPeek () override
+    {
+        return uart_peek_available(_uart);
+    }
+
+    // consume bytes after use (see peekBuffer)
+    void peekConsume (size_t consume) override
+    {
+        return uart_peek_consume(_uart, consume);
+    }
+
```

Comment:
@devyte, is this a spot for multiple inheritance?  Add a new `PeekBuffer` interface class?  Sorry if I'm getting all JAVA on you...

That said, I'm fine with this way, too.